resample/bootstrap.py

```diff
@@ -0,0 +1,285 @@
+    """
+    return np.asarray([fn(x) for x in resample(sample, size, **kwds)])
+
+
+def confidence_interval(
+    fn: Callable,
+    sample: np.ndarray,
+    cl: float = 0.95,
+    ci_method: str = "percentile",
+    **kwds,
+) -> Tuple[float, float]:
+    """
+    Calculate bootstrap confidence intervals.
+
+    Parameters
+    ----------
+    fn : callable
+        Function to be bootstrapped.
+    sample : array-like
+        Original sample.
+    cl : float, default : 0.95
+        Confidence level. Asymptotically, this is the probability that the interval
+        contains the true value.
+    ci_method : str, {'percentile', 'student', 'bca'}, optional
+        Confidence interval method. Default is 'percentile'.
+    **kwds
+        Keyword arguments forwarded to :func:`bootstrap`.
+
+    Returns
+    -------
+    (l, u) : tuple
+        Upper and lower confidence limits
+    """
+    if not 0 < cl < 1:
+        raise ValueError("cl must be between zero and one")
+
+    alpha = 1 - cl
+    thetas = bootstrap(fn, sample, **kwds)
+
+    if ci_method == "percentile":
+        return _confidence_interval_percentile(thetas, alpha / 2)
+
+    theta = fn(sample)
+
+    if ci_method == "student":
+        return _confidence_interval_studentized(theta, thetas, alpha / 2)
+
+    if ci_method == "bca":
+        j_thetas = _jackknife(fn, sample)
+        return _confidence_interval_bca(theta, thetas, j_thetas, alpha / 2)
+
+    raise ValueError(
+        "ci_method must be 'percentile', 'student', or 'bca', but "
+        f"'{ci_method}' was supplied"
+    )
+
+
+def _resample_stratified(sample, size, method, strata, rng):
+    # call resample on sub-samples and merge the replicas
+    sub_samples = [sample[strata == x] for x in np.unique(strata)]
+    for sub_replicas in zip(
+        *[resample(s, size, method=method, random_state=rng) for s in sub_samples]
+    ):
+        yield np.concatenate(sub_replicas, axis=0)
+
+
+def _resample_ordinary(
+    sample: np.ndarray, size: int, rng: np.random.Generator,
+) -> np.ndarray:
+    # i.i.d. sampling from empirical cumulative distribution of sample
+    n = len(sample)
+    for _ in range(size):
+        yield rng.choice(sample, size=n, replace=True)
+
+
+def _resample_balanced(
+    sample: np.ndarray, size: int, rng: np.random.Generator,
+) -> np.ndarray:
+    # effectively computes a random permutation of `size` concatenated
+    # copies of `sample` and returns `size` equal chunks of that
+    n = len(sample)
+    indices = rng.permutation(n * size)
+    for i in range(size):
+        sel = indices[i * n : (i + 1) * n] % n
+        yield sample[sel]
+
+
+def _fit_parametric_family(dist: stats.rv_continuous, sample: np.ndarray) -> Tuple:
+    if dist == stats.multivariate_normal:
+        # has no fit method...
+        return (np.mean(sample, axis=0), np.cov(sample.T, ddof=1))
+
+    if dist == stats.t:
+        fit_kwd = {"fscale": 1}  # HD: I think this should not be fixed to 1
```

Comment:
I think it would help to allow `fscale` to vary. Then one can use the parametric bootstrap with student's t also for distributions that are approximately normal, but have somewhat heavier tails. I found that the student's t distribution works well to approximately fit certain resonances. What do you think?

Reply:
I will remove the comment and rather add an issue to discuss this further, to move on with this PR.